src/dynamic_honey_badger/dynamic_honey_badger.rs

```diff
@@ -288,11 +290,16 @@ where
                 // If DKG completed, apply the change, restart Honey Badger, and inform the user.
                 debug!("{:?} DKG for {:?} complete!", self.our_id(), kgs.change);
                 self.netinfo = kgs.key_gen.into_network_info()?;
-                self.restart_honey_badger(batch_epoch + 1);
-                ChangeState::Complete(kgs.change)
+                self.restart_honey_badger(batch_epoch + 1, None);
+                ChangeState::Complete(Change::NodeChange(kgs.change))
             } else if let Some(change) = self.vote_counter.compute_winner().cloned() {
                 // If there is a new change, restart DKG. Inform the user about the current change.
-                step.extend(self.update_key_gen(batch_epoch + 1, &change)?);
+                step.extend(match &change {
+                    Change::NodeChange(change) => self.update_key_gen(batch_epoch + 1, &change)?,
+                    Change::EncryptionSchedule(schedule) => {
+                        self.update_encryption_schedule(batch_epoch + 1, *schedule)?
+                    }
+                });
```

Comment:
Shouldn't we go straight to `ChangeState::Complete` for this? Changes other than `NodeChange` doesn't need a key generation period—after the restart of Honey Badger, it's done—, so I wonder whether the `InProgress` variant should only take a `NodeChange` argument in the first place, instead of a `Change`?

Reply:
Probably.  We could certainly make that change, though I wouldn't be surprised about needing to reverse the change to `InProgress` later. I'll change the former, let me know if you feel strongly about the `InProgress` change.

Reply:
I don't think we'll ever have `InProgress` for another change: DKG only needs to run if we changed the set of validators, but not for any parameter changes. The only thing I can think of is that we may want to regenerate our keys periodically, even without a validator change, but we could just add a `NodeChange::NewKeys` for that later, if needed.

On the other hand, the API is possibly more intuitive if both variants contain the same `Change` type.

@vkomenda, @c0gent: I'm fine with either. Let me know if you have a preference.

Reply:
Note to self: changing this to `Complete` is what broke the tests.